src/irmin-pack/unix/pack_store.ml

```diff
@@ -124,9 +122,9 @@ struct
       Option.map (fun len -> min_length + len) t.size_of_value_and_length_header
   end
 
-  let read_and_decode_entry_prefix ~off ~io_read =
+  let read_and_decode_entry_prefix ~off ~io_read_at_most =
     let buf = Bytes.create Entry_prefix.max_length in
-    let bytes_read = io_read ~off buf in
+    let bytes_read = io_read_at_most ~off ~len:Entry_prefix.max_length buf in
     (* We may read fewer then [Entry_prefix.max_length] bytes when reading the
        final entry in the pack file (if the data section of the entry is
        shorter than [Varint.max_encoded_size]. In this case, an invalid read
```

Comment:
@tomjridge I think that this piece of code is a potential source of `read from gap` errors.

For example, if the entry being read is a blob of length 1 that is the last of a chunk in a prefix file, the above code will read after the entry (i.e. read in the gap).

Reply:
See also my comment in `io_read_and_decode_entry_prefix` below.

Reply:
I think we allow reading into a gap, so this is OK? (Allowing reading into a gap was motivated precisely by this pattern of "read more than we might need, because we don't know what the length is".) What is disallowed is to attempt to read starting from an offset in a gap. So I think this and the following `io_read_and_decode_entry_prefix` are OK? PROVIDED the object itself is live, we can read beyond the object (even if that involves reading into a gap).

Reply:
Ok I see. Thanks!